WordPress/openverse-frontend
# Replace Husky and lint-staged with pre-commit

## Fixes


Fixes #1641 by @sarayourfriend 

## Description


This PR replaces Husky and lint-staged with the tried-and-tested pre-commit that we're already using in all other repos.

This PR
- adds some new `pc:*` scripts to the `package.json` file to download, install and run pre-commit
- converts additional commands from the husky scripts to local repo hooks (`i18n`, `types`, `test:unit`)
- implements linting changes suggested by pre-commit

## Screenshots

See how the `test:unit` hook is skipped during the pre-commit staged but run during pre-push stage.

<img width="1392" alt="Screenshot 2022-09-25 at 2 13 13 PM" src="https://user-images.githubusercontent.com/16580576/192138604-4bdf7d38-6c36-4634-a220-688e13ef91aa.png">


## Testing Instructions


0. Checkout the PR. 
1. Undo Husky config `git config --unset core.hooksPath`
2. Make some code changes.
3. Make some valid changes, some invalid ones.
4. Try committing, see that hooks fail.
5. Fix the invalid changes.
6. Commit and see that hooks pass.
7. Push to see that the `test:unit` hook runs.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
